### PR TITLE
[WIP] merging points, polylines, ellipses, cuboid 2D and rotated boxes in exporting task annotation

### LINKED ISSUE #5634: Merging algorithm for 2D cuboids, polylines, points and rotated shapes is not implemented
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### PR DESCRIPTION
**Pull request by lengoanhcat**
### Motivation and context

Given two annotations `a` and `b` of the same geometric type (e.g. rectanges/boxes, cuboids 3D, cuboids 2D, polygons etc)  in two overlapping video segments annotated in two jobs under the same task, exporting the task annotation requires merging of `a` and `b` into one geometric object `c` if `a` and `b` are similar. 

These behaviors have been implemented for three geometric types
* ShapeType.RECTANGE
* ShapeType.CUBOID and DimensionType.DIM3D
* ShapeType.POLYGON

The aim of this PR is implementing and testing similar behaviors for 
* ShapeType.POINTS
* ShapeType.POLYLINE
* ShapeType.ELLIPSE
* ShapeType.CUBOID and DimensionType.DIM2D
* ShapeType.RECTANGE ( rotated boxes )

#### Similarity of ShapeType.POINTS

Give two points `a` and `b` with coordinates $(x_a, y_a)$ and $(x_b,y_b)$, I suggest usage of inverse of euclidean distance between two points: 
$\text{similarity}=\frac{1}{\sqrt{(x_a-x_b)^2+(y_a-y_b)^2}}$ 
This similarity will be in the range of $(0,+\infty)$

#### Similarity of ShapeType.POLYLINES [1]

Two possible similarity scores may be inverese of two types of distances for polylines i.e. Chamfer and Fretchet distances.

###

### How has this been tested?

I think automated tests should be implemented.

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)) 
  - Resolve #5634 
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### Refs

[1] Chamfer Distance - https://pdal.io/en/2.4.3/apps/chamfer.html

### CODE REVIEW

**Comment by zhiltsov-max** [comment]
Hi, it's a great contribution to the project! Please try adding corresponding tests in `cvat/apps/dataset_manager/tests/test_annotation.py`.

**Comment by nmanovic** [comment]
@lengoanhcat , thank you for the great contribution! Would you mind to help us finish the PR?
If you can add some tests, it will be great.
Please add a line into CHANGELOG.md. It will help others to see the important change.

File: cvat/apps/dataset_manager/annotation.py [hunk 1]
```diff
@@ -406,6 +431,21 @@ def _calc_polygons_similarity(p0, p1):
...
+                p1 = geometry.LineString(pairwise(obj1["points"]))
+
+                return _calc_linestrings_similarity(p0,p1)
+
+            elif obj0["type"] == ShapeType.ELLIPSE:
+                pass
+            elif obj0["type"] == ShapeType.CUBOID and dimension == DimensionType.DIM_2D:
+                pass
             else:
                 return 0 # FIXME: need some similarity for points, polylines, ellipses and 2D cuboids
```
Comment (nmanovic) [inline cvat/apps/dataset_manager/annotation.py:450]: Need to update the comment.

**Comment by nmanovic** [comment]
@lengoanhcat , It looks like for rotated bounding boxes and ellipses we can use `shapely` to calculate IoU and use it as the similarity metric.

Would you like to continue the patch?

**Comment by zhiltsov-max** [comment]
@nmanovic, both these cases should probably be compared as masks (quality control does it this way as well).

**Comment by bsekachev** [comment]
@lengoanhcat 

We will close the patch for now, please, let us know if you would like to continue it and we will reopen.

**Pull request closed without merging**